time ts
uint8[12608] buf
uint32 size